src/Agent/NewRelic/Agent/Extensions/Providers/Wrapper/NLogLogging/NLogWrapper.cs

```diff
@@ -68,28 +70,38 @@ private void DecorateLogMessage(object logEvent, Type logEventType, IAgent agent
                 return;
             }
 
-            var formattedMetadata = LoggingHelpers.GetFormattedLinkingMetadata(agent);
-
-            var messageGetter = _messageGetter ??= VisibilityBypasser.Instance.GeneratePropertyAccessor<string>(logEventType, "Message");
-
-            // Message should not be null, but better to be sure
-            var originalMessage = messageGetter(logEvent);
-            if (string.IsNullOrWhiteSpace(originalMessage))
+            var getFormattedMessageFunc = GetFormattedMessageFunc(logEventType);
+            var formattedMessage = getFormattedMessageFunc(logEvent);
+            if (string.IsNullOrWhiteSpace(formattedMessage))
             {
                 return;
             }
 
-            // this cannot be made a static since it is unique to each logEvent
-            var messageSetter = VisibilityBypasser.Instance.GeneratePropertySetter<string>(logEvent, "Message");
-            messageSetter(originalMessage + " " + formattedMetadata);
+            // NLog version strings are not setup to allow using min/max version in instrumentation - they only report major version.
+            // This will default to the 4.5+ field name and if that is not found, use the pre4.5 field name.
+            var formattedMessageName = FormattedMessage45Plus;
```

Comment:
I think this is a good solution. Should we have either a fallback or a try/catch in case they change the field name again?

Reply:
I don't think it will be needed.  It looks like they started following Microsoft's format, like we did, and that was the reason for the change.